### PR TITLE
#19077 Adding a Upgrade task to add the apps to the layout if does no…

### PR DESCRIPTION
**Pull request by jdotcms**
Adding the apps portlet to the layout if it does not already exists.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.UUIDUtil;
+import io.vavr.control.Try;
+
+import java.sql.SQLException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:12]: Codacy found an issue: [Avoid unused imports such as 'java.sql.SQLException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.Collections;
+import java.util.HashMap;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:14]: Codacy found an issue: [Avoid unused imports such as 'java.util.HashMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.db.DbConnectionFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    private void dropAppPortlet(final DotConnect dotConnect) {
+        try {
+            final String dropAppPortletSQL = "delete from cms_layouts_portlets where portlet_id = 'apps'";
+            dotConnect.executeStatement(dropAppPortletSQL);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:29]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test adding the APPS portlet to the layout
+ * @author jsanca
+ */
+public class Task05355AddAppsPortletToLayoutTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.UUIDUtil;
+import io.vavr.control.Try;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:16]: Codacy found an issue: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.db.DbConnectionFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:16]: Codacy found an issue: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:29]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:12]: Codacy found an issue: [Avoid unused imports such as 'java.sql.SQLException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:14]: Codacy found an issue: [Avoid unused imports such as 'java.util.HashMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:16]: Codacy found an issue: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:14]: Codacy found an issue: [Avoid unused imports such as 'java.util.HashMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:12]: Codacy found an issue: [Avoid unused imports such as 'java.sql.SQLException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test adding the APPS portlet to the layout
+ * @author jsanca
+ */
+public class Task537000AddAppsPortletToLayoutTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayout.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.db.DbConnectionFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    private void dropAppPortlet(final DotConnect dotConnect) {
+        try {
+            final String dropAppPortletSQL = "delete from cms_layouts_portlets where portlet_id = 'apps'";
+            dotConnect.executeStatement(dropAppPortletSQL);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:29]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.db.DbConnectionFactory'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:29]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.UUIDUtil;
+import io.vavr.control.Try;
+
+import java.sql.SQLException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java:12]: Codacy found an issue: [Avoid unused imports such as 'java.sql.SQLException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.Collections;
+import java.util.HashMap;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java:14]: Codacy found an issue: [Avoid unused imports such as 'java.util.HashMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.List;
+import java.util.Map;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task537000AddAppsPortletToLayout.java:16]: Codacy found an issue: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05355AddAppsPortletToLayoutTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task53700AddAppsPortletToLayoutTest.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test adding the APPS portlet to the layout
+ * @author jsanca
+ */
+public class Task53700AddAppsPortletToLayoutTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task53700AddAppsPortletToLayoutTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task53700AddAppsPortletToLayoutTest.java [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    private void dropAppPortlet(final DotConnect dotConnect) {
+        try {
+            final String dropAppPortletSQL = "delete from cms_layouts_portlets where portlet_id = 'apps'";
+            dotConnect.executeStatement(dropAppPortletSQL);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task53700AddAppsPortletToLayoutTest.java:29]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6018094)

**Pull request merged**